reflectx/reflect.go

```diff
@@ -6,20 +6,63 @@
 //
 package reflectx
 
-import "sync"
-
 import (
+	"fmt"
 	"reflect"
 	"runtime"
+	"strings"
+	"sync"
 )
 
-type fieldMap map[string][]int
+type field struct {
+	Index    []int
+	Path     string
+	Field    reflect.StructField
+	Zero     reflect.Value
+	Name     string
+	Options  map[string]string
+	Embedded bool
+}
+
+type fields struct {
+	Index []*field
+	Paths map[string]*field
+	Names map[string]*field
+}
+
+func (f fields) GetByPath(path string) *field {
```

Comment:
In what context would you use these paths?  If you are doing a marshaller or an unmarshaller, you'll probably be descending the object tree as you go, so you won't need to traverse in this manner.